core/trino-spi/src/main/java/io/trino/spi/shuffle/Shuffle.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.shuffle;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+public interface Shuffle
+        extends Closeable
+{
+    ShuffleHandle getHandle();
+
+    void outputTaskFinished(int outputTaskPartitionId);
+
+    void noMoreOutputTasks(Set<Integer> outputTaskPartitionIds);
```

Comment:
This method naming is weird as there are no methods like `addOutputTask`. Why not use `setOutputTasks`?

Reply:
Good point. Refactored the SPI based on the offline discussion we had.